src/SBInterpolatedImage.cpp

```diff
@@ -895,31 +881,61 @@ namespace galsim {
         }
         _flux = kValue(Position<double>(0.,0.)).real();
         dbg<<"flux = "<<_flux<<std::endl;
+        setCentroid();
+    }
+
+    void SBInterpolatedKImage::SBInterpolatedKImageImpl::setCentroid() const {
```

Comment:
Nice.  :)  Looks like a more useful optimization would be to let the user skip that sanity check if they know they are Hermitian.  But no need to do that here....